lib/api.js

```diff
@@ -48,6 +49,40 @@ function removeCoupling(heroku, app) {
            .then(coupling => deleteCoupling(heroku, coupling.id));
 }
 
+function getPipelineCouplings(heroku, pipelineId) {
+  return heroku.request({
+    method: 'GET',
+    path: `/pipelines/${pipelineId}/pipeline-couplings`,
+    headers: { 'Accept': V3_HEADER }
+  });
+}
+
+function getAppFitler(heroku, appIds) {
+  return heroku.request({
+    method: 'POST',
+    path: `/filters/apps`,
+    headers: { 'Accept': FILTERS_HEADER },
+    body: { in: { id: appIds } }
+  });
+}
+
+function getPipelineApps(heroku, pipelineId) {
+  return getPipelineCouplings(heroku, pipelineId).then((couplings) => {
+    const appIds = couplings.map((coupling) => coupling.app.id);
+
+    return getAppFitler(heroku, appIds).then((apps) => {
+      const couplingsByAppId = couplings.reduce((memo, coupling) => {
+        memo[coupling.app.id] = coupling;
+        return memo;
+      }, {});
+
+      apps.forEach((app) => app.coupling = couplingsByAppId[app.id]);
```

Comment:
Hah, so we've moved the stitching from API to here. Is this a pattern we'd like to retain, or do we want to refactor?